contracts/ERC20RewardApeV2.sol

```diff
@@ -208,16 +214,17 @@ contract ERC20RewardApeV1 is ReentrancyGuard, Ownable, Initializable {
         UserInfo storage user = userInfo[_user];
         updatePool();
         if (user.amount > 0) {
-            uint256 pending = (user.amount * poolInfo.accRewardTokenPerShare) /
+            uint256 pending = ((user.amount * poolInfo.accRewardTokenPerShare) /
                 1e30 -
-                user.rewardDebt;
+                user.rewardDebt) + user.pendingReward;
             if (pending > 0) {
-                // If rewardBalance is low then revert to avoid losing the user's rewards
-                require(
-                    rewardBalance() >= pending,
-                    "insufficient reward balance"
-                );
-                safeTransferRewardInternal(_user, pending, true);
+                uint256 rewardBal = rewardBalance();
+                if (pending > rewardBal) {
+                    user.pendingReward = pending - rewardBal;
+                    safeTransferRewardInternal(_user, rewardBal, true);
+                } else {
+                    safeTransferRewardInternal(_user, pending, true);
```

Comment:
Reset `user.pendingReward` to 0 here.
Once `user.pendingReward` is set to any number and enough rewards are there the user can just drain it right now.

Reply:
Nice catch, thanks! 🙏